### DISCUSSION TITLE
Can I preserve the variables during the trainning?

### DISCUSSION THREAD

**Discussion by Lastilla** [lululxvi/deepxde#990] [Q&A] [answered]
Just like the example in the invers about N-S Function, I set 2 variables:
`C1 = dde.Variable(0.0)`
`C2 = dde.Variable(0.0)`

And my pde:
`def Navier_Stokes_Equation(x, y):`
`    u = y[:, 0:1]`
`    v = y[:, 1:2]`
`    p = y[:, 2:3]`
`    du_x = dde.grad.jacobian(y, x, i=0, j=0)`
`    du_y = dde.grad.jacobian(y, x, i=0, j=1)`
`    dv_x = dde.grad.jacobian(y, x, i=1, j=0)`
`    dv_y = dde.grad.jacobian(y, x, i=1, j=1)`
`    dp_x = dde.grad.jacobian(y, x, i=2, j=0)`
`    dp_y = dde.grad.jacobian(y, x, i=2, j=1)`
`    du_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)`
`    du_yy = dde.grad.hessian(y, x, component=0, i=1, j=1)`
`    dv_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)`
`    dv_yy = dde.grad.hessian(y, x, component=1, i=1, j=1)`
`    continuity = du_x + dv_y`
`    x_momentum = C1 * (u * du_x + v * du_y) + dp_x - C2 * (du_xx + du_yy)`
`    y_momentum = C1 * (u * dv_x + v * dv_y) + dp_y - C2 * (dv_xx + dv_yy)`
`    return [continuity, x_momentum, y_momentum]`

I want to keep the variables during the trainning.

**Comment by lululxvi** [comment, marked as answer]
Don't understand "keep the variables during the trainning"
